### PR TITLE
[FLINK-2083] [docs] [ml]  Ensure high quality docs for FlinkML in 0.9

### PR DESCRIPTION
**Pull request by thvasilo**
Various fixes and additions to the docs for FlinkML

### CODE REVIEW

**Comment by thvasilo** [comment]
We could turn this into a WIP PR, and keep it open until 0.9 is released, in case we want to make more changes to the docs.

File: docs/libs/ml/optimization.md [hunk 1]
```diff
@@ -160,10 +158,20 @@ mini-batch SGD will be performed.
...
+          </p>
+        </td>
+      </tr>
+      <tr>
+        <td><strong>PredictionFunction</strong></td>
+        <td>
+          <p>
+            Class that provides the prediction function, used to calculate $\hat{y}$ based on the
+            weights and the example features, and the prediction gradient.
+            (Default value: <strong>LinearPrediction</strong>)
```
Comment (tillrohrmann) [inline docs/libs/ml/optimization.md]: Would be cool to have a link to a list of currently supported prediction functions.

File: docs/libs/ml/optimization.md [hunk 2]
```diff
@@ -184,12 +192,15 @@ mini-batch SGD will be performed.
...
           <p>
-            Class that provides the prediction function, used to calculate $\hat{y}$ based on the 
-            weights and the example features, and the prediction gradient.
-            (Default value: <strong>LinearPrediction</strong>)
+            When set, this value acts as a convergence stopping criterion for the algorithm.
+            If the relative difference in the objective function (loss) between two consecutive
+            iterations
+            is less than this value we stop the iterations and return the last weights.
+            This allows us to avoid unnecessary iterations after the algorithm has converged.
+            (Default value: <strong>None</strong>)
```
Comment (tillrohrmann) [inline docs/libs/ml/optimization.md]: Maybe we could specify the convergence criterion also as a formula, then it's clear which value is used to calculate the relative change. Maybe we can make a list with all supported convergence schemes (at the moment only the relative change one).

File: docs/libs/ml/optimization.md (hunk 1 above)
Comment (tillrohrmann) [inline docs/libs/ml/optimization.md]: There we could also specify the math of the prediction function.

Comment (tillrohrmann) [inline docs/libs/ml/optimization.md]: The same applies to the loss functions and the regularization functions. I know they are already described in the text, but maybe it is helpful to have again a more doc-style listing of all available options with a description.

**Comment by tillrohrmann** [comment]
Looks good modulo some minor comments.

I think we can merge this PR once the comments are addressed. We can then simply open new PRs for new changes.

File: docs/libs/ml/optimization.md (hunk 1 above)
Comment (thvasilo) [inline docs/libs/ml/optimization.md]: Should I add these in their own table then? i.e. one for prediction functions, one for loss functions etc.

File: docs/libs/ml/optimization.md (hunk 2 above)
Comment (thvasilo) [inline docs/libs/ml/optimization.md]: As above, create a separate table?

File: docs/libs/ml/optimization.md (hunk 1 above)
Comment (tillrohrmann) [inline docs/libs/ml/optimization.md]: Yes that would be great, IMO.

File: docs/libs/ml/optimization.md (hunk 2 above)
Comment (tillrohrmann) [inline docs/libs/ml/optimization.md]: That would be great IMO

**Comment by tillrohrmann** [comment]
LGTM. Will merge it.

**Pull request closed without merging**